develop-guide/writeparser/writeparser.rst

```diff
@@ -1,47 +1,558 @@
+
 .. _write-parser:
 
-Write a parser
-======================
-This topic describes why and how to write your own parsers to meet your network automation requirements.
+##############
+Write a parser
+##############
+
+This topic describes why and how to write your own parsers to meet your network automation requirements.
+
+Remember to share your new parser with the rest of the |pyATS| user community! Please see the topic :ref:`contribute` for more information.
+
+.. tip::
+    If you're the kind of person who prefers to learn by video
+    instead of reading about it, then please checkout these great videos
+    about making pyATS parsers:
+
+    - `How to write a Genie parser for Cisco! <https://youtu.be/ibLNilSfdTc>`_ - uploaded by `Juhi Mahajan <https://www.youtube.com/channel/UCA0EsjQEEJac9XIidC6v6Rg>`_
+        - Juhi provides a good overview of parser creation process
+    - `Creating a pyATS | Genie Parser from SCRATCH <https://youtu.be/knxkbWTamBY>`_ - uploaded by `Data Knox <https://www.youtube.com/channel/UCi7SD3zfCjkiDWvSFthIQSg>`_
+        - Data Knox's gives a full hands on parser writing guide from start to finish
+
+This guide contains lots information and exmaples,
+so please feel free to use the following links to jump around in this guide as
+you choose.
+
+| 1. :ref:`What is a parser? <what_is_a_parser>`
+| 2. :ref:`Why write a parser? <why_write_a_parser>`
+| 3. :ref:`How the pyATS Library parsers work <how_the_parsers_work>`
+| 4. :ref:`Setting up your development environment <setting_up_your_environment>`
+| 5. :ref:`Writing the Schema Class <writing_the_schema_class>`
+|   5.1 :ref:`Creating a schema <creating_a_schema>`
+|     5.1.1 :ref:`Identifying keys directy from show command output <keys_from_show_command>`
+|     5.1.2 :ref:`Identifying keys from XML output <keys_from_XML>`
+|     5.1.3 :ref:`Identifying keys from the YANG data model <keys_from_YANG>`
+|   5.2 :ref:`Explore other pyATS schemas <schemaexamples>`
+| 6. :ref:`Writing the Parser Class <writing_the_parser_class>`
+
+TODO
+Finish this ^^ table of contents
+
+.. _what_is_a_parser:
+
+********************
+1. What is a parser?
+********************
+
+.. include:: ../definitions/def_parser.rst
+    :start-line: 3
+
+* For a basic introduction to the |library| parsers, see the topic `Parse device output <https://pubhub.devnetcloud.com/media/pyats-getting-started/docs/quickstart/parseoutput.html#parse-device-output>`_ in the *Get Started with pyATS* guide (recommended).

```

Comment:
At least give 1 summary sentence

"A parser converts device text output into a structured Python dictionary. With this dictionary, automation becomes much easier and much more powerful. "

Something like that

Reply:
There's something there, it's just stored in `def_parser.rst`

![image](https://user-images.githubusercontent.com/33130026/109345696-8800ee80-783e-11eb-868c-b1576ce9fa29.png)